src/index.ts

```diff
@@ -162,10 +207,65 @@ const invalidateDeployedPullRequest = async (
   }
 };
 
+const updateOutdatedDeployment = async (
+  context: Context<Webhooks.WebhookPayloadPush>
+) => {
+  const contextBranch = context.payload.ref.replace("refs/heads/", "");
+  const { preProductionEnvironment } = config;
+  const deployment = await findDeployment(context, preProductionEnvironment);
+  let deployedPr;
+
+  if (!deployment) {
+    debug(`No deployment found for ${preProductionEnvironment} - quitting`);
+    return;
+  }
+
+  try {
+    const deployedPrNumber = deploymentPullRequestNumber(deployment);
+
+    deployedPr = (
+      await context.github.pulls.get(
+        context.repo({ pull_number: deployedPrNumber })
+      )
+    ).data;
+  } catch (ex) {
+    // move on
+  }
+
+  if (!deployedPr) {
+    debug(
+      `Could not find PR associated with ${preProductionEnvironment} deployment - quitting`
+    );
+    return;
+  }
+
+  const prBranch = deployedPr.head.ref;
+
+  if (prBranch !== contextBranch) {
+    debug(
+      `Push is unrelated to ${preProductionEnvironment} deployment - nothing to do (${prBranch} vs ${contextBranch})`
+    );
+    return;
+  }
+
+  debug(
+    `Re-deploying ${deployedPr.number} to ${preProductionEnvironment} with new commits...`
+  );
+
+  return Promise.all([
+    setCommitStatus(context, deployedPr, "pending"),
+    handleQA(context, deployedPr),
+  ]);
+};
+
 const probot = (app: Application) => {
   // Additional app.on events will need to be added to the `on` section of the example workflow in README.md
   // https://help.github.com/en/actions/reference/events-that-trigger-workflows
 
+  app.on(["push"], async (context) => {
```

Comment:
You could use `pull_request.synchronized` instead of `push` and avoid needing to fetch a pull request inside of `updateOutdatedDeployment` (you could do something like `context.payload.pull_request.number === deployedPr`) and to reduce the number of times this gets run (and perhaps reinstate the `master` branch check in the workflow's `if` statement).

Reply:
If you have just one event use
```suggestion
  app.on("push", async (context) => {
```
so that `context` gets proper type for that event.